### ISSUE TITLE
Add support for RIP-7212

### ISSUE THREAD

**Issue by Amxx** [NomicFoundation/edr#523]
### Describe the feature

RIP-7212 provides a precompile, at address(0x100) for secp256r1 verification. This is already live on multiple networks, including Matic (polygon).

It is currently not possible to test code that relies on this precompile using hardhat.

Some pointers:
- [REVM has this precompile implemented, but only loads it for FJORD](https://github.com/bluealloy/revm/blob/main/crates/revm/src/optimism/handler_register.rs#L148-L153), this would require some change so that it can be activated manually through the config
- This option will then have to be exposed through the hardhat config.

### Search terms

secp265r1, p256, precompile, RIP-7212, EIP-7212

**Comment by fvictorio** [comment]
@alcuadrado any preference for the config option? I'd suggest:

```
networks: {
  hardhat: {
    enableRip7212: true, // defaults to false
```

Some alternatives:
- `enableRIP7212`
- `enableSecp265r1`
- `enableSecp265r1Precompile`

**Comment by alcuadrado** [comment]
I think it's better to name the standard, because that's more precise than just naming the curve

**Comment by fvictorio** [comment]
To unit test this:

A call to the precompile address with this calldata:

```
0x4cee90eb86eaa050036147a12d49004b6b9c72bd725d39d4785011fe190f0b4da73bd4903f0ce3b639bbbf6e8e80d16931ff4bcf5993d58468e8fb19086e8cac36dbcd03009df8c59286b162af3bd7fcc0450c9aa81be5d10d312af6c66b1d604aebd3099c618202fcfe16ae7770b0c49ab5eadf74b754204a3bb6060e44eff37618b065f9832de4ca6ca971a7a1adc826d0f7c00181a5fb2ddf79ae00b4e10e
```

should return `0x0000000000000000000000000000000000000000000000000000000000000001`.

[Source](https://github.com/mds1/evm-diff/pull/65), which in turn is copied from [here](https://github.com/maticnetwork/bor/blob/bade7f57df5c09ae060c15fc66aed488c526149e/core/vm/testdata/precompiles/p256Verify.json).

**Comment by Wodann** [comment]
I added support for RIP-7212 in EDR and exposed a config option to Hardhat. On their end, they need to expose it to the end user.

**Comment by fvictorio** [comment]
@Amxx we just released [Hardhat v2.22.7](https://github.com/NomicFoundation/hardhat/releases/tag/hardhat%402.22.7), which includes a new [`enableRip7212`](https://hardhat.org/hardhat-network/docs/reference#enablerip7212) option.

You can enable it like this:

```
networks: {
  hardhat: {
    enableRip7212: true,
  },
},
```
